src/redux/stock/actionCreator.js

```diff
@@ -0,0 +1,40 @@
+import { API_URL, API_KEY } from 'react-native-dotenv';
+
+import { FETCH_STOCK_DETAILS_START, FETCH_STOCK_DETAILS_FAIL, FETCH_STOCK_DETAILS_SUCCESS } from './actionType';
+
+const fetchStockDetailsStart = () => ({
+  type: FETCH_STOCK_DETAILS_START,
+});
+
+const fetchStockDetailsFail = (err) => ({
+  type: FETCH_STOCK_DETAILS_FAIL,
+  payload: err,
+});
+
+const fetchStockDetailsSuccess = (data) => ({
+  type: FETCH_STOCK_DETAILS_SUCCESS,
+  payload: data,
+});
+
+// eslint-disable-next-line import/prefer-default-export
+export const fetchStockDetails = (symbol) => (dispatch) => {
+  const url = new URL(`${API_URL}/stock/${symbol}/advanced-stats`);
+  url.searchParams.append('token', API_KEY);
+  dispatch(fetchStockDetailsStart());
+  fetch(url)
+    .then((res) => {
+      if (!res.ok) {
+        switch (res.status) {
+          case 404:
+            throw new Error(`The stock ${symbol} you are looking for does not exist.`);
+          case 403:
+            throw new Error('Permission denied.');
```

Comment:
Do you want to display this error message to the user? This could be misleading. Of course it is permission denied, but the app has no right to access the data, not the user. It is like the 500 on the backend side, we made some mistake, not the user.

Reply:
Updated. Now we are only handling 404 as a special case and the rest will be `"Oops, there\'s something wrong with our app."`